lib/Parser/RegexParser.cpp

```diff
@@ -3014,9 +3014,17 @@ namespace UnifiedRegex
         program->source[bodyChars] = 0;
         program->sourceLen = bodyChars;
 
-        program->numGroups = nextGroupId;
-
-        AssertOrFailFast(0 < program->numGroups && program->numGroups <= MAX_NUM_GROUPS);
+        // We expect nextGroupId to be positive, because the full regexp itself always
+        // counts as a capturing group.
+        Assert(nextGroupId > 0);
+        if (nextGroupId > MAX_NUM_GROUPS)
+        {
+            Js::JavascriptError::ThrowSyntaxError(this->scriptContext, JSERR_RegExpTooManyCapturingGroups);
```

Comment:
Is this a RangeError?

Reply:
That might make more sense, but it's not really a RangeError in the sense the spec usually uses it, I think. The user might think "what range?" when encountering this, although the error message is helpful in this case.

Reply:
^ to make clear my opinion on this: "meh" -- either type of error is fine IMO since this Error is not in spec (spec assumes infinite memory etc)

Reply:
Talked to @boingoing, who convinced me that, although neither choice is obviously The Right Thing, RangeError is probably more consistent with the rest of the system.

Reply:
I also don't feel too strongly about the error type but we do use RangeError to indicate that a bound function call has too many arguments so this case is sort of consistent with that.